### PR TITLE
[1.4] Fix NPCID.Sets.MPAllowedEnemies ignoring modded entries + ExampleMod adjustment

### PR DESCRIPTION
**Pull request by direwolf420**
### Description
Fixes a hardcoded NPCID.Count value to use NPCID.Sets.MPAllowedEnemies.Length instead to make it see modded entries.

Also adds new 1.4 "boss summon" item code by adjusting current "Plantera Item" in ExampleMod.

### CODE REVIEW

File: ExampleMod/Content/Items/Consumables/PlanteraItem.cs [hunk 1]
```diff
@@ -6,13 +6,23 @@
...
+			ItemID.Sets.SortingPriorityBossSpawns[Type] = 12; // This helps sort inventory know that this is a boss summoning Item.
+
+			//This is set to true for all NPCs that can be summoned via an Item (calling NPC.SpawnOnPlayer). If this is for a modded boss,
+			//write this in the bosses file instead
+			NPCID.Sets.MPAllowedEnemies[NPCID.Plantera] = true;
+		}
+
+		public override void Unload() {
+			//If the mod is disabled, the flag should be reset again to retain the original state of the game
+			NPCID.Sets.MPAllowedEnemies[NPCID.Plantera] = false;
```
Comment (Chicken-Bones) [inline ExampleMod/Content/Items/Consumables/PlanteraItem.cs:25]: @Mirsario should tML be resetting this array? I guess we don't revert the effects of GlobalX on Sets arrays normally. 
Wait.. we re-run the static constructors which should reset them yes?

Comment (Chicken-Bones) [reply ExampleMod/Content/Items/Consumables/PlanteraItem.cs:25]: @direwolf420 please test if this is required.

**Review by Chicken-Bones — CHANGES_REQUESTED** [review]

File: ExampleMod/Content/Items/Consumables/PlanteraItem.cs (hunk 1 above)
Comment (Mirsario) [reply ExampleMod/Content/Items/Consumables/PlanteraItem.cs:25]: Yes, all sets get reset, unloading isn't needed.

Comment (direwolf420) [reply ExampleMod/Content/Items/Consumables/PlanteraItem.cs:25]: Yes, just tested, printing false when ExampleMod is unloaded again, will fix

**Pull request merged**